limitador-server/src/main.rs

```diff
@@ -227,47 +244,335 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
-                                Err(e) => error!("Failed reloading limit file: {}", e),
-                            }
-                        });
+    let limiter = Arc::clone(&rate_limiter);
+    let handle = Handle::current();
+
+    let mut watcher = RecommendedWatcher::new(move |result: Result<Event, Error>| match result {
+        Ok(ref event) => {
+            if let EventKind::Modify(ModifyKind::Data(_)) = event.kind {
+                let limiter = limiter.clone();
+                let location = event.paths.first().unwrap().clone();
+                handle.spawn(async move {
+                    match limiter.load_limits_from_file(&location).await {
+                        Ok(_) => info!("Reloaded limit file"),
+                        Err(e) => error!("Failed reloading limit file: {}", e),
                     }
-                }
-                Err(ref e) => {
-                    warn!("Something went wrong while watching limit file: {}", e);
-                }
-            })?;
+                });
+            }
+        }
+        Err(ref e) => {
+            warn!("Something went wrong while watching limit file: {}", e);
+        }
+    })?;
 
-        watcher.watch(Path::new(&limits_file_path), RecursiveMode::Recursive)?;
-        Some(watcher)
-    } else {
-        None
-    };
+    watcher.watch(Path::new(&limit_file), RecursiveMode::Recursive)?;
 
     info!("Envoy RLS server starting on {}", envoy_rls_address);
     tokio::spawn(run_envoy_rls_server(
         envoy_rls_address.to_string(),
         rate_limiter.clone(),
     ));
 
+    info!("HTTP server starting on {}", http_api_address);
     run_http_server(&http_api_address, rate_limiter.clone()).await?;
 
     Ok(())
 }
+
+fn create_config() -> Configuration {
+    // figure defaults out
+    let default_limit_file = env::var("LIMITS_FILE").unwrap_or_else(|_| "".to_string());
+
+    let default_rls_host =
+        env::var("ENVOY_RLS_HOST").unwrap_or_else(|_| Configuration::DEFAULT_IP_BIND.to_string());
+    let default_rls_port =
+        env::var("ENVOY_RLS_PORT").unwrap_or_else(|_| Configuration::DEFAULT_RLS_PORT.to_string());
+
+    let default_http_host =
+        env::var("HTTP_API_HOST").unwrap_or_else(|_| Configuration::DEFAULT_IP_BIND.to_string());
+    let default_http_port =
+        env::var("HTTP_API_PORT").unwrap_or_else(|_| Configuration::DEFAULT_HTTP_PORT.to_string());
+
+    let redis_url = env::var("REDIS_URL").unwrap_or_else(|_| "".to_string());
+
+    let redis_cached_ttl_default = env::var("REDIS_LOCAL_CACHE_MAX_TTL_CACHED_COUNTERS_MS")
+        .unwrap_or_else(|_| (DEFAULT_MAX_TTL_CACHED_COUNTERS_SEC * 1000).to_string());
+    let redis_flushing_period_default = env::var("REDIS_LOCAL_CACHE_FLUSHING_PERIOD_MS")
+        .unwrap_or_else(|_| (DEFAULT_FLUSHING_PERIOD_SEC * 1000).to_string());
+    let redis_max_cached_counters_default = DEFAULT_MAX_CACHED_COUNTERS.to_string();
```

Comment:
This wasn't configurable until now... I did not add it to the env vars tho... should we?

Reply:
:shrug: 

create issue and somebody can implement